main.go

```diff
@@ -1687,48 +1690,137 @@ func recordedAsNotObfuscated(obj types.Object) bool {
 	return ok
 }
 
-func (tf *transformer) removeUnnecessaryImports(file *ast.File) {
-	usedImports := make(map[string]bool)
-	ast.Inspect(file, func(n ast.Node) bool {
-		node, ok := n.(*ast.Ident)
-		if !ok {
-			return true
-		}
+func (tf *transformer) resolveImportSpec(imp *ast.ImportSpec) *types.Package {
+	obj, ok := tf.info.Implicits[imp]
+	if !ok {
+		obj = tf.info.Defs[imp.Name] // renamed or dot import
+	}
+	if obj == nil {
+		return nil
+	}
 
-		uses, ok := tf.info.Uses[node]
-		if !ok {
-			return true
-		}
+	pkgObj, ok := obj.(*types.PkgName)
+	if !ok {
+		panic(fmt.Sprintf("unexpected object type for %s: %v", imp.Path.Value, obj))
+	}
+	return pkgObj.Imported()
+}
 
-		if pkg := uses.Pkg(); pkg != nil {
-			usedImports[pkg.Path()] = true
+// isSafeForInstanceType returns true if the passed type is safe for var declaration
+// Unsafe types: generic structs, funcs or interfaces, type constraint
+func isSafeForInstanceType(typ types.Type) bool {
+	switch t := typ.(type) {
+	case *types.Named:
+		if t.TypeParams().Len() > 0 {
+			return false
 		}
+		return isSafeForInstanceType(t.Underlying())
+	case *types.Signature:
+		return t.TypeParams().Len() == 0
+	case *types.Interface:
+		return t.IsMethodSet()
+	}
+	return true
+}
 
-		return true
-	})
-
+func (tf *transformer) makeImportsUsed(file *ast.File) {
 	for _, imp := range file.Imports {
 		if imp.Name != nil && imp.Name.Name == "_" {
 			continue
 		}
 
-		path, err := strconv.Unquote(imp.Path.Value)
-		if err != nil {
-			panic(err)
+		pkg := tf.resolveImportSpec(imp)
+		if pkg == nil {
+			panic(fmt.Sprintf("import %s not found", imp.Path.Value))
 		}
 
-		// The import path can't be used directly here, because the actual
-		// path resolved via go/types might be different from the naive path.
-		lpkg, err := listPackage(path)
-		if err != nil {
-			panic(err)
-		}
+		generated := false
+		scope := pkg.Scope()
+		for _, name := range scope.Names() {
```

Comment:
nit: iterate from 0 to N-1 with scope.Len and scope.Child, then you don't need to collect all the names and use Lookup.

Reply:
`Child()` returns Scope, not declarations